clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/ServerGroupManagerController.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2017 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.netflix.spinnaker.clouddriver.controllers;
+
+import com.netflix.spinnaker.clouddriver.model.ServerGroupManager;
+import com.netflix.spinnaker.clouddriver.model.ServerGroupManagerProvider;
+import com.netflix.spinnaker.clouddriver.requestqueue.RequestQueue;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.prepost.PostAuthorize;
+import org.springframework.security.access.prepost.PostFilter;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Slf4j
+@RestController
+@RequestMapping("/applications/{application}/serverGroupManagers")
+public class ServerGroupManagerController {
+  final List<ServerGroupManagerProvider> serverGroupManagerProviders;
+
+  final RequestQueue requestQueue;
+
+  @Autowired
+  public ServerGroupManagerController(List<ServerGroupManagerProvider> serverGroupManagerProviders, RequestQueue requestQueue) {
+    this.serverGroupManagerProviders = serverGroupManagerProviders;
+    this.requestQueue = requestQueue;
+  }
+
+  @PreAuthorize("hasPermission(#application, 'APPLICATION', 'READ')")
+  @PostFilter("hasPermission(filterObject.account, 'ACCOUNT', 'READ')" )
+  @RequestMapping(method = RequestMethod.GET)
+  Set<ServerGroupManager> getForApplication(@PathVariable String application) {
```

Comment:
A server group manager's server groups might not all live in the same application, right? I wonder how the UI should show that...

Reply:
That's fine, if you're on a certain "cluster" tab you would only see the server groups in that app. Clicking on the server group manager could show you the list of owned server groups (even ones in different apps) in the details panel the same way we do for load balancers.
![ygauhxgpjwl](https://user-images.githubusercontent.com/4874941/32517710-38b2e49e-c3d5-11e7-876f-ebde093be51c.png)

Reply:
Ah are you worried about server groups who's manager isn't in the current app? It's probably safe to ignore those (at least for now).

Reply:
We could also annotate server groups with their managers, and if one points to a manager not in this app, we can add a little button like the load balancer button, but this feels like an unlikely use-case.